src/components/Checkbox/Checkbox.js

```diff
@@ -25,11 +27,53 @@ type CheckboxProps = {
   onBlur?: (?SyntheticFocusEvent<HTMLInputElement>) => void,
 }
 
-class Checkbox extends PureComponent<CheckboxProps> {
+type CheckboxState = {
+  iconName: 'Check' | 'Minus',
```

Comment:
@lesha1201 is it really needed to store the icon name in the state? Could we just select it according to the `indeterminate` and `checked` props?

Reply:
@zouxuoz I did because we use transition for icon, so I need to keep previous icon. If we select it according to the `indeterminate` and `checked`, like this:
```jsx
<Icon name={indeterminate ? 'Minus' : 'Check'} />
```
and `indeterminate` changes from `true` to `false`, then we will see minus icon then check icon and then nothing

If we do it like this:
```jsx
const iconName = indeterminate ? 'Minus'
                               : checked ? 'Checked' : null;
return (<>{iconName && <Icon name={iconName} />}</>);
```
and `indeterminate` or `checked` changes from `true` to `false`, then we won't see any transition

Reply:
@lesha1201 oh, good point